### DISCUSSION TITLE
add text watermark (possibly with 2 different fonts), Magick 7

### DISCUSSION THREAD

**Discussion by Carl873** [ImageMagick/ImageMagick#5827] [Help] [answered]
Hi, despite having researched (googled) this, and despite having perfectly fine running code to add an image watermark, I haven't' been able to add a text watermark to an image.

Here is how I add the image watermark (in PS)

```
  $MagickArguments = @(
    $sourceImage.FullName,
    $strip, $Watermark,
    '-gravity',$Gravity,
    '-define',('compose:args=' + $Opacity + ',100'),
    '-compose',
    'dissolve',
    '-composite',
    $resultImage
  )
  magick $MagickArguments
```

But how can I add a text watermark, in its simplest form?

The next challenge is, how could I make the watermark using two different fonts, if possible, in one step? I could imagine doing it in two steps, but it would not be ideal.

In two steps, suppose the watermark test is "MyWatermark", I might add "My        " in one font, and then and then "  Watermark" in a second step. But like I said, it would not be ideal.

**Comment by snibgo** [comment, marked as answer]
Some examples of multiple fonts on one line of an image are at https://imagemagick.org/Usage/text/#mixed_font_lines . An alternative is to use "pango:", on the same page.

**Comment by Carl873** [comment]
Thank you! For some reason, I did not find this. I only found examples with the old command syntax (convert instead of magick).
